TLM/TLM/Util/TaskList.cs

```diff
@@ -0,0 +1,241 @@
+        /// Guaranteed to exist.
+        /// </param>
+        /// <remarks>
+        /// The event handlers must at least mark items active or inactive,
+        /// as applicable to the event.
+        /// </remarks>
+        public delegate void TaskListEvent(int index);
+
+        /// <summary>
+        /// Delegate for mandatory <see cref="TaskIsActive"/> method which is
+        /// used to test wheter an item in the <see cref="Tasks"/> array is
+        /// marked as active.
+        /// </summary>
+        /// <param name="index">
+        /// The index of the item in the <see cref="Tasks"/> array.
+        /// Guaranteed to exist when invoked by <see cref="TaskList{T}"/>.
+        /// </param>
+        /// <returns>
+        /// Return <c>true</c> if the item is active, othewise <c>false</c>.
+        /// </returns>
+        public delegate bool TaskListItemActive(int index);
+
+        /// <summary>
+        /// The <see cref="Tasks"/> array will be increased in multiples
+        /// of this size when necessary.
+        /// </summary>
+        /// <remarks>
+        /// Defaults to <see cref="BLOCK_SIZE"/> can be modified via ctor.
+        /// </remarks>
+        public int BlockSize { get; private set; }
+
+        /// <summary>
+        /// Array for storing tasks.
+        /// </summary>
+        /// <remarks>
+        /// Similar to <c>FastList.m_buffer</c>.
+        /// </remarks>
+        public T[] Tasks { get; private set; }
+
+        /// <summary>
+        /// Current number of used elements in <see cref="Tasks"/> array.
+        /// </summary>
+        /// <remarks>
+        /// Similar to <c>FastList.m_size</c>.
+        /// </remarks>
+        public int Size { get; private set; }
+
+        /// <summary>
+        /// Mandatory. Called after an item is added to the <see cref="Tasks"/>
+        /// array at specified <c>index</c>.
+        /// </summary>
+        public TaskListEvent OnAddTask;
+
+        /// <summary>
+        /// Mandatory. Called after an item is removed from the <see cref="Tasks"/>
+        /// array at specified <c>index</c>.
+        /// </summary>
+        public TaskListEvent OnRemoveTask;
+
+        /// <summary>
+        /// Mandatory. Invoked when <see cref="TaskList{T}"/> needs to
+        /// check whether a task at specified <c>index</c> is active.
+        /// </summary>
+        public TaskListItemActive TaskIsActive;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TaskList{T}"/> class.
+        /// </summary>
+        /// <param name="blockSize">
+        /// Optional: Specify custom block size. Defaults to <see cref="BLOCK_SIZE"/>.
+        /// Must be Power of 2.
+        /// </param>
+        public TaskList(int blockSize = BLOCK_SIZE) {
+            Shortcuts.AssertPowerOf2(blockSize);
+
+            BlockSize = blockSize;
+        }
+
+        /// <summary>
+        /// Returns the unused capacity of the <see cref="Tasks"/> array.
```

Comment:
`Capacity` means full storage size before we need to reallocate, i'd expect `TasksCapacity` to return allocated length of `Tasks`. `Available` or `AvailableStorage` are maybe better name candidates?

Reply:
It's badly named, I'll rename to something like `RemainingCapacity` ?